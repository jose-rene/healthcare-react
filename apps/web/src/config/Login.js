const {
    REACT_APP_INACTIVITY_TIMEOUT = 10,
    REACT_APP_LOGOUT_COUNTDOWN_TIME = 5,
} = process.env;

export const INACTIVITY_TIMEOUT = +REACT_APP_INACTIVITY_TIMEOUT;
export const LOGOUT_COUNTDOWN_TIME = +REACT_APP_LOGOUT_COUNTDOWN_TIME;
