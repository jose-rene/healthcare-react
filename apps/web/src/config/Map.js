const {
    MAP_API_KEY = "AIzaSyCYHqP5sGELeYR6eTPunrWgz1p6ctQ2DHs",
    MAP_BASE_URL = "https://maps.googleapis.com/maps/api/geocode/json",
} = process.env;

export const API_KEY = MAP_API_KEY;
export const BASE_URL = MAP_BASE_URL;
