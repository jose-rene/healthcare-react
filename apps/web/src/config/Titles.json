{
    "Mr.": "Mr.",
    "Ms.": "Ms.",
    "Mrs.": "Mrs.",
    "Miss.": "Miss.",
    "Dr.": "Dr.",
    "Sir": "Sir"
}
