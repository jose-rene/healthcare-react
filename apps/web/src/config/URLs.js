const {
    REACT_APP_API_HOST = "127.0.0.1",
    REACT_APP_API_PORT = "8000",
} = process.env;

const API_PROTOCOL = REACT_APP_API_HOST.indexOf("http") === 0 ? "" : "http://";
export const BASE_URL = `${API_PROTOCOL}${REACT_APP_API_HOST}:${REACT_APP_API_PORT}`;
export const API_URL = `${BASE_URL}/v1`;
export const LOGIN_URL = `${BASE_URL}/oauth/token`;
export const USER_URL = `${API_URL}/user`;
export const AUTH_TOKEN_NAME = "@dme.login.access_token";
export const HTTP_TIMEOUT = 10000;

export const POST = "post";
export const PUT = "put";
export const GET = "get";
export const DELETE = "delete";
