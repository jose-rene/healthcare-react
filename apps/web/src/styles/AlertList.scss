.box-alerts {
    margin: 0;
    width: 100%;
    display: flex;
    background: #ffffff;
    box-shadow: 0px 30px 40px rgba(0, 0, 0, 0.04);
    border-radius: 6px;
    padding: 26px;
  }
  
  .checkbox-label {
    font-style: normal;
    font-weight: 600;
    font-size: 14px;
    line-height: 17px;
    margin-left: 10px;
    color: #475866;
  }
  
  .checkbox-time {
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 17px;
    text-align: right;
    color: #475866;
    opacity: 0.8;
    margin: 0;
    margin-top: 4px;
  }
  
  .checkbox-subtitle {
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 17px;
    padding: 0;
    margin-left: 25px;
    color: #475866;
    opacity: 0.8;
  }
  
  .white-box-alerts {
    background: #ffffff;
    box-shadow: 0px 30px 40px rgba(0, 0, 0, 0.04);
    border-radius: 6px;
    padding: 0.5rem 0.5rem 0;
  }
  
  .container-button-alerts {
    display: flex;
    justify-content: flex-end;
  }

  .text-select-all {
    font-style: normal;
    font-weight: normal;
    font-size: 14px;
    line-height: 17px;
    text-align: right;
    color: #2e94e6;
    margin-bottom: 32px;
    margin-top: 16px;
  }
