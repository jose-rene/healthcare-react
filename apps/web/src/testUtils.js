// test-utils.js
import React from "react";
import { Router, Route } from "react-router-dom";
import { createMemoryHistory } from "history";
import { render as rtlRender } from "@testing-library/react";
import { createStore, applyMiddleware } from "redux";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import axios from "axios";
import MockAdapter from "axios-mock-adapter";
import rootReducer from "./reducers/index";

/**
 * render
 *
 * custom render with redux function
 * */
function render(
    ui,
    {
        initialState,
        store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>;
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

function renderWithRouter(
    ui,
    {
        route = "/",
        history = createMemoryHistory({ initialEntries: [route] }),
        initialState,
        store = createStore(rootReducer, initialState, applyMiddleware(thunk)),
        ...renderOptions
    } = {}
) {
    function Wrapper({ children }) {
        return (
            <Router history={history}>
                <Route path="/">
                    <Provider store={store}>{children}</Provider>
                </Route>
                <Route
                    path="/access-denied"
                    render={() => <div>Denied Stub</div>}
                    exact
                />
                <Route
                    path="/dashboard"
                    render={() => <div>Dashboard Stub</div>}
                />
            </Router>
        );
    }
    return {
        ...rtlRender(ui, { wrapper: Wrapper, history, ...renderOptions }),
        history,
    };
}

const axiosMock = () => {
    return new MockAdapter(axios);
};

// re-export everything
export * from "@testing-library/react";
// override render method, pass function to mock axios
export { render, renderWithRouter, axiosMock };
