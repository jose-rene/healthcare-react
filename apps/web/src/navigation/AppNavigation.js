import React, { useEffect } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";

import { useUser } from "Context/UserContext";

import { ADMIN } from "actions/types";

import PrivateRoute from "route/PrivateRoute";
import RoleRouteRouter from "route/RoleRoute";

import useApiCall from "hooks/useApiCall";

import RequestEdit from "pages/healthplan/RequestEdit";
import Account from "pages/Account/Account";
import Table from "pages/Test/Table";
import HpAddUser from "pages/healthplan/AddUser";
import HpEditUser from "pages/healthplan/EditUser";
import HpAddMember from "pages/healthplan/AddMember";
import HpSearchMember from "pages/healthplan/SearchMember";
import HpTraining from "pages/Training/Training";
import NewRequestAdd from "pages/newRequestAdd/NewRequestAdd";
import RequestLookup from "pages/RequestLookup/RequestLookup";
import AssessmentList from "pages/Assessment/AssessmentList";
import Assessment from "pages/Assessment/Assessment";
import Error401 from "pages/Errors/401";
import Federated from "pages/Federated";
import ForgotPassword from "pages/ForgotPassword";
import Login from "pages/Login";
import Error from "pages/NotFound";
import SetForgotPassword from "pages/SetForgotPassword";
import NotificationList from "pages/NotificationList";
import AdminPayer from "pages/Test/AdminPayer";
import AdminCompanies from "pages/Admin/Companies/Companies";
import AdminAddCompanies from "pages/Admin/Companies/AddCompanies";
import AdminDetailCompanies from "pages/Admin/Companies/DetailCompanies";
import AdminClinicians from "pages/Admin/Clinicians/Clinicians";
import AdminAddClinicians from "pages/Admin/Clinicians/AddClinicians";
import FormView from "pages/Admin/FormBuilder/show";
import FormBuilderEdit from "pages/Admin/FormBuilder/edit";
import FormIndex from "pages/Admin/FormBuilder/Index";
import AdminUserList from "pages/Admin/UserList/UserList";
import AdminUserEdit from "pages/Admin/UserList/UserEdit";
import AdminUserAdd from "pages/Admin/UserList/UserAdd";
import FormWizard from "pages/FormWizard";
import RequestSections from "pages/RequestSections";
import RequestSectionsShowForm from "pages/RequestSections/RequestSectionsShowForm";
import NarrativeReport from "../pages/Admin/NarrativeReport";
import EditNarrativeReport from "../pages/Admin/NarrativeReport/Edit";

const AppNavigation = () => {
    const [{ loading }, fireInitializeUser] = useApiCall({
        url: "user/profile",
    });

    const { initUser, getUser, logout } = useUser();
    const { initializing } = getUser();

    useEffect(() => {
        /**
         * on page load call the api to refresh the user state
         */
        let isMounted = true;
        if (isMounted) {
            (async () => {
                try {
                    const response = await fireInitializeUser();
                    // initializeUser(response);
                    initUser(response);
                } catch (e) {
                    console.log("fail!!!");
                    logout();
                }
            })();
        }
        // cleanup
        return () => {
            isMounted = false;
        };
        // eslint-disable-next-line react-hooks/exhaustive-deps
    }, []);

    if (loading || initializing) {
        return null;
    }

    return (
        <BrowserRouter>
            <Switch>
                <Route path="/" component={Login} exact />
                <Route path="/sso" component={Federated} />
                <Route path="/password/reset" component={ForgotPassword} />
                <Route path="/password/change" component={SetForgotPassword} />
                <RoleRouteRouter path="/dashboard" component="Home" />
                <PrivateRoute
                    path="/access-denied"
                    component={Error401}
                    exact
                />
                <PrivateRoute path="/account" component={Account} />
                <PrivateRoute
                    path="/healthplan/addmember"
                    middleware={["hp_champion", "hp_user"]}
                    component={HpAddMember}
                />
                <PrivateRoute
                    path="/healthplan/start-request"
                    middleware={["hp_manager", "hp_champion", "hp_user"]}
                    component={HpSearchMember}
                />
                <PrivateRoute
                    path="/healthplan/adduser"
                    middleware={["hp_champion", "create-users"]}
                    component={HpAddUser}
                />

                <PrivateRoute
                    path="/healthplan/user/:id"
                    middleware={["hp_champion", "create-users"]}
                    component={HpEditUser}
                />
                <PrivateRoute
                    path="/member/:member_id/request/add"
                    middleware={["hp_user", "hp_champion"]}
                    component={NewRequestAdd}
                />
                <PrivateRoute
                    path="/member/:member_id/request/:request_id/edit"
                    middleware={["hp_user", "hp_manager", "hp_champion"]}
                    component={NewRequestAdd}
                />
                <PrivateRoute
                    path="/requests/:request_id/form-sections/:form_slug"
                    middleware={[
                        "hp_user",
                        "hp_manager",
                        "hp_champion",
                        "coo",
                        "client_services_specialist",
                        "client_services_manager",
                    ]}
                    component={RequestSectionsShowForm}
                />
                <PrivateRoute
                    path="/requests/:request_id/form-sections"
                    middleware={[
                        "hp_user",
                        "hp_manager",
                        "hp_champion",
                        "coo",
                        "client_services_specialist",
                        "client_services_manager",
                    ]}
                    component={RequestSections}
                />
                <PrivateRoute
                    path="/requests/:id"
                    middleware={[
                        "hp_user",
                        "hp_manager",
                        "hp_champion",
                        "coo",
                        "client_services_specialist",
                        "client_services_manager",
                    ]}
                    component={RequestEdit}
                />
                <PrivateRoute
                    path="/healthplan/requests"
                    middleware={["hp_user", "hp_manager", "hp_champion"]}
                    component={RequestLookup}
                />
                <PrivateRoute
                    path="/healthplan/training"
                    middleware={["hp_user", "hp_champion"]}
                    component={HpTraining}
                />
                <PrivateRoute
                    exact
                    path="/admin/test/table"
                    middleware={[ADMIN]}
                    page={<Table />}
                />
                <PrivateRoute
                    exact
                    path="/admin/test/payer"
                    requiredAbility="can:edit-payer"
                    page={<AdminPayer />}
                />

                <PrivateRoute
                    path="/admin/forms/:form_slug/edit"
                    middleware={["software_engineer"]}
                    component={FormBuilderEdit}
                />

                <PrivateRoute
                    path="/forms/:form_slug/show"
                    middleware={["software_engineer"]}
                    component={FormView}
                />

                <PrivateRoute
                    path="/form-wizard/:form_group_slug"
                    middleware={[]}
                    component={FormWizard}
                />

                <PrivateRoute
                    path="/admin/forms"
                    middleware={["software_engineer"]}
                    component={FormIndex}
                />

                <PrivateRoute
                    path="/admin/narrative-report/:slug/edit"
                    middleware={["software_engineer"]}
                    component={EditNarrativeReport}
                    addLayout
                />

                <PrivateRoute
                    path="/admin/narrative-report"
                    middleware={["software_engineer"]}
                    component={NarrativeReport}
                    addLayout
                />

                <PrivateRoute
                    path="/admin/companies"
                    middleware={["software_engineer"]}
                    component={AdminCompanies}
                />

                <PrivateRoute
                    path="/admin/add-companies"
                    middleware={["software_engineer"]}
                    component={AdminAddCompanies}
                />

                <PrivateRoute
                    path="/admin/company/:id"
                    middleware={["software_engineer"]}
                    component={AdminDetailCompanies}
                />

                <PrivateRoute
                    path="/admin/company/:id/edit"
                    middleware={["software_engineer"]}
                    component={AdminDetailCompanies}
                />

                <PrivateRoute
                    exact
                    path="/admin/clinicians"
                    middleware={["software_engineer"]}
                    component={AdminClinicians}
                />

                <PrivateRoute
                    path="/admin/add-clinicians"
                    middleware={["software_engineer"]}
                    component={AdminAddClinicians}
                />

                <PrivateRoute
                    path="/admin/clinicians/:id/edit"
                    middleware={["software_engineer"]}
                    component={AdminAddClinicians}
                />

                <PrivateRoute
                    exact
                    path="/admin/users"
                    middleware={["create-users"]}
                    component={AdminUserList}
                />

                <PrivateRoute
                    path="/admin/users/add"
                    middleware={["create-users"]}
                    component={AdminUserAdd}
                />

                <PrivateRoute
                    path="/admin/users/:id/edit"
                    middleware={["create-users"]}
                    component={AdminUserEdit}
                />

                <PrivateRoute
                    path="/notifications"
                    component={NotificationList}
                />

                <PrivateRoute
                    path="/admin/assessments"
                    middleware={["software_engineer"]}
                    component={AssessmentList}
                />

                <PrivateRoute path="/assessment/:id" component={Assessment} />

                {/* ERROR PAGES */}
                <Route component={Error} />
            </Switch>
        </BrowserRouter>
    );
};

export default AppNavigation;
