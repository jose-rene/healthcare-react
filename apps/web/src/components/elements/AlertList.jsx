import { debounce } from 'lodash';
import React, { useCallback, useEffect, useMemo, useState } from 'react';
import { ListGroup, ListGroupItem } from 'react-bootstrap';
import { PUT } from '../../config/URLs';
import useApiCall from '../../hooks/useApiCall';
import '../../styles/AlertList.scss';
import Icon from './Icon';

const subjectTypeMap = ({ type: notificationType = '' }) => {
    switch (true) {
        case (notificationType.match(/RequestActivity/)?.index >= 0):
            return 'Activity Request';
        default:
            return 'General Notification';
    }
};

const List = () => {
    const [{ loading, data = [] }, fireLoadNotifications] = useApiCall({
        url: 'notifications',
        defaultData: [],
    });
    const [{}, fireMarkRead] = useApiCall({
        method: PUT,
        url: 'notifications',
    });
    const debouncedFireLoad = debounce(fireLoadNotifications, 1000);
    const [alerts, setAlerts] = useState([]);
    const [alertsLoading, setAlertsLoading] = useState({});

    const enableCheckAll = useMemo(() => {
        return alerts.filter(a => !a.read_at).length > 0;
    }, [alerts]);

    const reloadNotifications = useCallback(() => {
        fireLoadNotifications();
    }, [fireLoadNotifications]);

    const handleCheckAll = useCallback(async () => {
        if (!enableCheckAll) {
            return false;
        }
        const ids = alerts.filter(({ read_at }) => !read_at);
        const allIds = ids.map(({ id }) => id);
        ids.forEach(({ id }) => {
            setAlertsLoading({ ...alertsLoading, [id]: true });
        });
        await fireMarkRead({ params: { ids: allIds } });
        await fireLoadNotifications();

        setAlertsLoading({});
    }, [fireLoadNotifications]);

    const handleCheckChange = async ({ target: { name, checked: value = false } }) => {
        // if not checked don't do anything
        if (!value) {
            return false;
        }

        setAlertsLoading({ ...alertsLoading, [name]: true });

        await fireMarkRead({
            params: { ids: [name] },
        });

        // Wait for a second and if no other checks are checked then reload alerts
        await debouncedFireLoad();

        // after firing the api call to get alerts set the checked alerts to not loading to clean itself up.
        setTimeout(() => {
            setAlertsLoading({ ...alertsLoading, [name]: false });
        }, 1010);
    };

    /**
     * When new data is loaded populate the alerts state
     */
    useEffect(() => {
        setAlerts(data);
    }, [data]);

    /**
     * on load get notifications
     */
    useEffect(() => {
        reloadNotifications();
    }, []);

    return (
        <>
            <div className="box-same-line">
                <h1 className="box-title" onClick={debouncedFireLoad}>
                    Alerts {loading && (
                    <Icon size="1x" icon={'spinner'} spin/>
                )}
                </h1>
                <div className="d-none d-sm-block">
                    {enableCheckAll && (
                        <a href={void (0)}
                           aria-disabled={enableCheckAll}
                           onClick={handleCheckAll}
                           className={`btn-link text-select-all ${enableCheckAll
                               ? ''
                               : 'not-allowed'}`}>
                            Select All
                        </a>
                    )}
                </div>
            </div>
            <div className="white-box-alerts">
                <div className="container-items">
                    <ListGroup data-testid="alert-list-group">
                        {alerts && alerts.map(({ id, read_at, type, data, human_created_at }) => (
                            <ListGroupItem className="border-0" key={id} data-testid={`alert-${id}`}>
                                <div className="d-flex">
                                    <div>
                                        {alertsLoading[id] ? (
                                            <Icon size="1x" icon="spinner"
                                                  spin/>
                                        ) : (
                                            <input
                                                id={id}
                                                name={id}
                                                onChange={handleCheckChange}
                                                type="checkbox"
                                                value={true}
                                                checked={!!read_at}
                                                disabled={!!read_at}
                                            />)}
                                    </div>
                                    <div>
                                        <label
                                            htmlFor={id}
                                            className={`checkbox-label ${read_at ? 'text-muted' : ''}`}>
                                            {subjectTypeMap({ type })}
                                        </label>
                                    </div>
                                    <div className="ml-auto">
                                        <p className="checkbox-time">{human_created_at}</p>
                                    </div>
                                </div>
                                <p className="checkbox-subtitle">
                                    {data.message}
                                </p>
                            </ListGroupItem>
                        ))}
                    </ListGroup>
                </div>
            </div>
        </>
    );
};

export default List;
