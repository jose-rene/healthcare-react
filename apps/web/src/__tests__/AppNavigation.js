import AsyncStorage from '@react-native-community/async-storage';
import '@testing-library/jest-dom';
import { generate as generateRandomString } from 'randomstring';
import React from 'react';
import { AUTH_TOKEN_NAME } from '../config/URLs';
import AppNavigation from '../navigation/AppNavigation';
import { axiosMock, fireEvent, profileResponse, render, screen, wait } from '../testUtils';
import { notificationResponse } from './AlertList';

// mock the window location
const initialLocation = window.location;
delete window.location;

beforeEach(() => {
    window.location = {
        ...initialLocation,
        reload: jest.fn(),
        assign: jest.fn(),
    };
});
afterEach(() => {
    window.location = initialLocation;
    axiosMock().reset();
});

describe("App Navigation", () => {
    it("renders login with redux state defaults", async () => {
        // render with redux
        render(<AppNavigation />, {
            userToken: generateRandomString({
                length: 24,
                charset: "alphanumeric",
            }),
            isLoading: false,
        });
        // wait for the state changes
        const login = await screen.findByRole("button", { name: /sign in/i });
        // expect to see the page
        expect(login).toBeTruthy();
    });
    it("redirects from dashboard to login when not authenticated", async () => {
        // render with redux
        render(<AppNavigation />, {
            isLoading: false,
            route: "/dashboard",
        });
        // wait for the state changes
        const login = await screen.findByRole("button", { name: /sign in/i });
        // expect to see the page
        expect(login).toBeTruthy();
    });
    it("displays access denied when authenticated without role", async () => {
        // with feature added this will make an api request, mock axios here
        const profileResponseNoRole = {
            ...profileResponse,
            roles: [],
            primary_role: "",
        };
        axiosMock().onGet(/profile/).reply(200, profileResponseNoRole);
        const authToken = generateRandomString({
            length: 24,
            charset: "alphanumeric",
        });
        await AsyncStorage.setItem(AUTH_TOKEN_NAME, authToken);
        // render with redux
        render(<AppNavigation />, {
            isLoading: false,
            route: "/dashboard",
        });
        await wait(() =>
            expect(window.location.assign).toHaveBeenCalledWith(
                "/access-denied"
            )
        );
    });
    it("displays dashboard when authenticated with role", async () => {
        axiosMock().onGet(/profile/).reply(200, profileResponse).onGet(/inspire/).reply(200, {
            message: 'Fly a kite in a thunderstorm. - Benjamin Franklin',
        }).onGet(/summary/).reply(200, {
            new: 33,
            in_progress: 12,
            scheduled: 20,
            submitted: 22,
        }).onGet(/notifications/).reply(200, notificationResponse);

        const authToken = generateRandomString({
            length: 24,
            charset: "alphanumeric",
        });
        await AsyncStorage.setItem(AUTH_TOKEN_NAME, authToken);
        // render with redux
        render(<AppNavigation />, {
            isLoading: false,
            route: "/dashboard",
        });

        // wait for the state changes
        const dash = await screen.findByRole("button", { name: /search/i });
        // expect to see the dash page
        expect(dash).toBeTruthy();
        const userInfo = await screen.getByTestId("userinfo");
        expect(userInfo).toBeTruthy();
        // wait for the state changes
        // const welcome = await screen.getByRole("heading", {
        //    level: 1,
        //    name: "Welcome to your Portal",
        // });
        // expect to see the dash page
        // await wait(() => expect(welcome).toBeTruthy());
    });
    it("logs the user out", async () => {
        // render with redux
        axiosMock().onGet(/profile/).reply(200, profileResponse).onGet(/inspire/).reply(200, {
            message: 'Fly a kite in a thunderstorm. - Benjamin Franklin',
        }).onGet(/summary/).reply(200, {
            new: 33,
            in_progress: 12,
            scheduled: 20,
            submitted: 22,
        }).onGet(/notifications/).reply(200, notificationResponse);

        const authToken = generateRandomString({
            length: 24,
            charset: "alphanumeric",
        });
        await AsyncStorage.setItem(AUTH_TOKEN_NAME, authToken);
        // render with redux
        render(<AppNavigation />, {
            isLoading: false,
            route: "/dashboard",
        });

        // wait for the state changes
        const dash = await screen.findByRole("button", { name: /search/i });
        // expect to see the dash page
        expect(dash).toBeTruthy();
        const link = await screen.findByRole("link", { name: /logout/i });
        expect(link).toBeTruthy();
        // click on logout
        fireEvent.click(link);
        await wait(async () =>
            expect(window.location.reload).toHaveBeenCalled()
        );
    });
});
