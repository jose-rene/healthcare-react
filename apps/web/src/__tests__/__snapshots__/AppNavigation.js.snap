// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert List Displays an alert list with read and unread alerts 1`] = `
<div
  class="list-group"
  data-testid="alert-list-group"
>
  <div
    class="border-0 list-group-item"
    data-testid="alert-88362598-0027-4ac1-8950-b6b1d407dc3d"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          id="88362598-0027-4ac1-8950-b6b1d407dc3d"
          name="88362598-0027-4ac1-8950-b6b1d407dc3d"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label "
          for="88362598-0027-4ac1-8950-b6b1d407dc3d"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          5 seconds ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Aut mollitia eos et dolores possimus nobis dolor.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-cd3d1a9f-1765-418e-9ded-834f07fd7d2a"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          id="cd3d1a9f-1765-418e-9ded-834f07fd7d2a"
          name="cd3d1a9f-1765-418e-9ded-834f07fd7d2a"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label "
          for="cd3d1a9f-1765-418e-9ded-834f07fd7d2a"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          7 seconds ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Quidem ullam voluptates vel similique a.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-3f8a6c03-e48d-4418-aaa2-16d26a8ce1f9"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          id="3f8a6c03-e48d-4418-aaa2-16d26a8ce1f9"
          name="3f8a6c03-e48d-4418-aaa2-16d26a8ce1f9"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label "
          for="3f8a6c03-e48d-4418-aaa2-16d26a8ce1f9"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          8 seconds ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Possimus hic voluptas aperiam reprehenderit fuga est id.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-2334ce4c-d14c-420b-91e4-23f997ce2a3b"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          id="2334ce4c-d14c-420b-91e4-23f997ce2a3b"
          name="2334ce4c-d14c-420b-91e4-23f997ce2a3b"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label "
          for="2334ce4c-d14c-420b-91e4-23f997ce2a3b"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          12 seconds ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Consequatur dolorum ipsa nostrum consequuntur architecto.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-3f909940-866c-43e3-a10f-024ebfda3542"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          checked=""
          disabled=""
          id="3f909940-866c-43e3-a10f-024ebfda3542"
          name="3f909940-866c-43e3-a10f-024ebfda3542"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label text-muted"
          for="3f909940-866c-43e3-a10f-024ebfda3542"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          22 minutes ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Officiis dolorem aut ipsum quasi veritatis explicabo.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-2e5bb3d2-fb51-4710-8b5d-730124034340"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          checked=""
          disabled=""
          id="2e5bb3d2-fb51-4710-8b5d-730124034340"
          name="2e5bb3d2-fb51-4710-8b5d-730124034340"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label text-muted"
          for="2e5bb3d2-fb51-4710-8b5d-730124034340"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          22 minutes ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Eius quis consequatur ullam et autem expedita.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-196ab37f-a55e-421b-a72d-1087563dfa4d"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          checked=""
          disabled=""
          id="196ab37f-a55e-421b-a72d-1087563dfa4d"
          name="196ab37f-a55e-421b-a72d-1087563dfa4d"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label text-muted"
          for="196ab37f-a55e-421b-a72d-1087563dfa4d"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          23 minutes ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Magni iste neque accusantium magnam dicta voluptatum voluptatibus.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-851a3ff8-b1db-433e-923f-b153d6014d6b"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          checked=""
          disabled=""
          id="851a3ff8-b1db-433e-923f-b153d6014d6b"
          name="851a3ff8-b1db-433e-923f-b153d6014d6b"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label text-muted"
          for="851a3ff8-b1db-433e-923f-b153d6014d6b"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          23 minutes ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Incidunt temporibus tenetur nobis dolorem magnam.
    </p>
  </div>
  <div
    class="border-0 list-group-item"
    data-testid="alert-8281feb5-c00a-49fd-8db2-b0243e3ecca3"
  >
    <div
      class="d-flex"
    >
      <div>
        <input
          checked=""
          disabled=""
          id="8281feb5-c00a-49fd-8db2-b0243e3ecca3"
          name="8281feb5-c00a-49fd-8db2-b0243e3ecca3"
          type="checkbox"
          value="true"
        />
      </div>
      <div>
        <label
          class="checkbox-label text-muted"
          for="8281feb5-c00a-49fd-8db2-b0243e3ecca3"
        >
          Activity Request
        </label>
      </div>
      <div
        class="ml-auto"
      >
        <p
          class="checkbox-time"
        >
          1 hour ago
        </p>
      </div>
    </div>
    <p
      class="checkbox-subtitle"
    >
      Iure est et alias rem.
    </p>
  </div>
</div>
`;
